definitions/marts/looker_studio_marts/ecf/ls_ecf_21_24_manual_adjustment.sqlx

```diff
@@ -0,0 +1,196 @@
+    }
+}
+
+-- Gather 2021 declaration type totals per lead provider & participant type to create a banding qualifier.
+--- The funding band applied to a declaration is dependant on the number of applicants with the same declaration type with the lead provider.
+--- This CTE calculates how many declarations by type were paid to Lead providers in the 2021 statement cohort.
+--- This will be used to assess the newly raised declarations for rolled over participants to provide a banding in 2021.
+WITH bandings_2021 AS (
+  SELECT
+     declarations.cpd_lead_provider_name
+    ,declarations.declaration_type
+    ,COUNT(DISTINCT declarations.declaration_id) AS banding_qualifier
+  FROM
+    ${ref('ecf_declarations')} AS declarations
+  WHERE
+    declarations.cpd_lead_provider_name IS NOT NULL
+    AND
+    (
+    -- No further changes are being made to the 2021 statement cohort so only paid declarations will be valid. Any in an eligible or payable state will need to be reraised or the participant to be transferred.
+      (declarations.state = 'paid' AND declarations.statement_cohort = 2021)
+      OR
+      (declarations.funded_declaration = TRUE AND declarations.cohort_rolled_over_21_24 = TRUE)
+    )
+  GROUP BY
+     declarations.cpd_lead_provider_name
+    ,declarations.declaration_type
+),
+
+-- Create table of upper particiapnt volume targets taken from the live service for 2021.
+--- In the 2021 statement/contract a Band D existed where if an LP exceeded their upper limit. In subsequent contracts this banding was removed.
+--- The CTE below creates a table of these upper limits for each LP to be used in evaluating the banding qualifier created above.
+targets_2021 AS (
+  SELECT
+     'Ambition Institute' AS cpd_lead_provider_name
+    ,10001 AS target
+  UNION ALL
+  SELECT
+     'Best Practice Network' AS cpd_lead_provider_name
+    ,6001 AS target
+  UNION ALL
+  SELECT
+     'Education Development Trust' AS cpd_lead_provider_name
+    ,7126 AS target
+  UNION ALL
+  SELECT
+     'Teach First' AS cpd_lead_provider_name
+    ,8801 AS target
+  UNION ALL
+  SELECT
+     'UCL Institute of Education' AS cpd_lead_provider_name
+    ,8958 AS target
+),
+
+-- Gather 2024 statement cohort declarations and create 2024 banding qualifier values.
+--- This CTE looks at all funded declarations (e/p/p) in the 2024 statement cohort and produces a count by LP & Statement Date (date of payment)
+--- This will be used to assess the newly raised declarations for rolled over participants to provide a banding in 2024.
+bandings_2024 AS (
+  SELECT
+     PARSE_DATE('%B %Y', statements.name) AS statement_date
+    -- This logic assumes that there will only be one statement_date in the future as when a declaration is submitted it is tagged against the next statement period and not any further in time.
+    ,CASE 
+      WHEN PARSE_DATE('%B %Y', statements.name) > CURRENT_DATE() THEN TRUE
+      ELSE FALSE
+     END AS is_current_statement

```

Comment:
Where is this used?

Reply:
This flag is used in the dashboard as a pre-set filter so that the user doesn't need to apply it manually each time they load it.

Reply:
Can you highlight below where it's pulling through? Sorry if I'm being a bit thick but I can't see where the field from this CTE pulls through below.

Reply:
Apologies, I'd left a test case statement in the final output rather than using this field. Will update and push shortly.